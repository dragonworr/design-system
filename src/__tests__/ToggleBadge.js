import React from 'react'
import renderer from 'react-test-renderer'
import { ToggleBadge, theme } from '..'

describe('ToggleBadge', () => {
  test('selected one has light blue background-color', () => {
    const json = renderer.create(<ToggleBadge selected />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.light)
    expect(json).toHaveStyleRule('color', theme.colors.darkBlue)
  })

  test('unselected one has white background-color and hover state', () => {
    const json = renderer.create(<ToggleBadge />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.white)
    expect(json).toHaveStyleRule('color', theme.colors.darkBlue)
  })
})
