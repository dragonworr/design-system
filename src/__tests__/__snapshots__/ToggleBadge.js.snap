// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleBadge selected ToggleBadge renders with default props 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
}

.c1:disabled {
  opacity: 0.25;
}

.c1:hover {
  background-color: #049;
}

.c0 {
  border-radius: 2px;
  display: inline-block;
  font-weight: 600;
  background-color: #cdf;
  color: #049;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
}

.c0:hover {
  background-color: #cdf;
}

<button
  className="c0 c1"
  color="darkBlue"
  fontSize={0}
  selected={true}
/>
`;

exports[`ToggleBadge selected one with background-color and text color passed as props hover state 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
}

.c1:disabled {
  opacity: 0.25;
}

.c1:hover {
  background-color: #049;
}

.c0 {
  border-radius: 2px;
  display: inline-block;
  font-weight: 600;
  background-color: #0a0;
  color: #c00;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 14px;
}

.c0:hover {
  background-color: #0a0;
}

<button
  className="c0 c1"
  color="red"
  fontSize={1}
  selected={true}
/>
`;

exports[`ToggleBadge unselected ToggleBadge renders with default props 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
}

.c1:disabled {
  opacity: 0.25;
}

.c1:hover {
  background-color: #049;
}

.c0 {
  border-radius: 2px;
  display: inline-block;
  font-weight: 600;
  background-color: #fff;
  color: #049;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  font-size: 12px;
}

.c0:hover {
  background-color: #cdf;
}

<button
  className="c0 c1"
  color="darkBlue"
  fontSize={0}
  selected={false}
/>
`;
